src/main/java/org/cyclonedx/maven/DefaultModelConverter.java

```diff
@@ -172,21 +174,43 @@ public Component convert(Artifact artifact, CycloneDxSchema.Version schemaVersio
         if (CycloneDxSchema.Version.VERSION_10 != schemaVersion) {
             component.setBomRef(component.getPurl());
         }
-        if (isDescribedArtifact(artifact)) {
-            try {
-                final MavenProject project = getEffectiveMavenProject(artifact);
-                if (project != null) {
-                    extractComponentMetadata(project, component, schemaVersion, includeLicenseText);
-                }
-            } catch (ProjectBuildingException e) {
-                if (logger.isDebugEnabled()) {
-                    logger.warn("Unable to create Maven project for " + artifact.getId() + " from repository.", e);
-                } else {
-                    logger.warn("Unable to create Maven project for " + artifact.getId() + " from repository.");
-                }
+
+        // Read the project
+        MavenProject project = null;
+        try {
+            project = getEffectiveMavenProject(artifact);
+        } catch (ProjectBuildingException error) {
+            if (logger.isDebugEnabled()) {
+                logger.warn("Unable to create Maven project for `{}` from repository.", artifact.getId(), error);
+            } else {
+                logger.warn("Unable to create Maven project for `{}` from repository.", artifact.getId());
+            }
+        }
+
+        if (project != null) {
+
+            // Populate external references
+            setExternalReferences(component, externalReferences);
+
+            // Extract the rest of the metadata for JARs, i.e., *described* artifacts
+            if (isDescribedArtifact(artifact)) {
+                extractComponentMetadata(project, component, schemaVersion, includeLicenseText);
```

Comment:
then all this refactoring is not necessary

Reply:
Fixed in 20287587e8123fabf00afdacf6ab1cba581459be.